samples/net/cloud/tagoio_http_post/src/sockets.c

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2020 Gerson Fernando Budke <nandojve@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(tagoio_http_post, CONFIG_TAGOIO_HTTP_POST_LOG_LEVEL);
+
+#include <net/net_ip.h>
+#include <net/socket.h>
+#include <net/socketutils.h>
+#include <net/dns_resolve.h>
+#include <net/tls_credentials.h>
+#include <net/http_client.h>
+
+#include "sockets.h"
+
+#if IS_ENABLED(CONFIG_TAGOIO_MANUAL_SERVER)
+#define TAGOIO_SERVER CONFIG_TAGOIO_API_SERVER_IP
+#else
+#define TAGOIO_SERVER "api.tago.io"
+#endif
+
+#define TAGOIO_URL          "/data"
+#define HTTP_PORT              "80"
+#define HTTPS_PORT            "443"
+
+static const char *tagoio_http_headers[] = {
+	"Device-Token: " CONFIG_TAGOIO_DEVICE_TOKEN "\r\n",
+	"Content-Type: application/json\r\n",
+	"_ssl: false\r\n",
+	NULL
+};
+
+int tagoio_connect(struct tagoio_context_t *ctx)
+{
+	struct addrinfo *addr;
+	struct addrinfo hints;
+	char hr_addr[INET6_ADDRSTRLEN];
+	char *port;
+	int dns_attempts = 3;
+	int ret = -1;
+
+	memset(&hints, 0, sizeof(hints));
+
+	hints.ai_socktype = SOCK_STREAM;
+	hints.ai_protocol = IPPROTO_TCP;
+	if (IS_ENABLED(CONFIG_NET_IPV6)) {
+		hints.ai_family = AF_INET6;
+	} else if (IS_ENABLED(CONFIG_NET_IPV4)) {
+		hints.ai_family = AF_INET;
+	}
+
+	if (IS_ENABLED(CONFIG_NET_SOCKETS_SOCKOPT_TLS)) {
+		port = HTTPS_PORT;
```

Comment:
HTTPS support doesn't seem to be implemented in the sample, this line could be misleading for users. I'd stick to the HTTP_PORT.

Reply:
Agreed!